.gitattributes

```diff
@@ -0,0 +1,65 @@
+###############################
+# Git Line Endings            #
+###############################
+
+# Set the default behavior, in case people don't have core.autocrlf set.
+* text=auto eol=lf
+
+# Explicitly declare text files you want to always be normalized and converted
+# to native line endings on checkout.
+*.txt text
+*.yml text
+*.py text
+*.cmake text
+*.c text
+*.cpp text
+*.h text
+*.hpp text
+.gitattributes text
+.gitignore text
+.dockerignore text
+*.md text
+.clang-format text
+.clang-tidy text
+Dockerfile text
+*.Dockerfile text
+Dockerfile.* text
+LICENSE text
+
+# Force batch scripts to always use CRLF line endings so that if a repo is accessed
+# in Windows via a file share from Linux, the scripts will work.
+*.{cmd,[cC][mM][dD]} text eol=crlf
+*.{bat,[bB][aA][tT]} text eol=crlf
+
+# Force bash scripts to always use LF line endings so that if a repo is accessed
+# in Unix via a file share from Windows, the scripts will work.
+*.sh text eol=lf
+
+###############################
+# Git Large File System (LFS) #
+###############################
+
+# Archives
+*.7z filter=lfs diff=lfs merge=lfs -text
+*.br filter=lfs diff=lfs merge=lfs -text
+*.gz filter=lfs diff=lfs merge=lfs -text
+*.tar filter=lfs diff=lfs merge=lfs -text
+*.zip filter=lfs diff=lfs merge=lfs -text
+
+# Documents
+*.pdf filter=lfs diff=lfs merge=lfs -text
+
+# Images
+*.gif filter=lfs diff=lfs merge=lfs -text
+*.ico filter=lfs diff=lfs merge=lfs -text
+*.jpg filter=lfs diff=lfs merge=lfs -text
+*.pdf filter=lfs diff=lfs merge=lfs -text
+*.png filter=lfs diff=lfs merge=lfs -text
+*.psd filter=lfs diff=lfs merge=lfs -text
+*.webp filter=lfs diff=lfs merge=lfs -text
+
+# Fonts
+*.woff2 filter=lfs diff=lfs merge=lfs -text
+
+# Other
+*.exe filter=lfs diff=lfs merge=lfs -text
```

Comment:
I don't mean to nitpick, but I am not sure what adding a `.gitattributes` file has to do with this feature. I think that this could easily be a separate PR. Is this necessary for CMakePresets/VSCode support?

Reply:
There was a problem with the line endings. When I was developing inside the Docker the VSCode used LF because it was in a Linux OS but when I was developing in VSCode and native windows it was using CRLF. So git diff showed that the whole file had changed.